stable/datadog/templates/cluster-agent-deployment.yaml

```diff
@@ -116,7 +116,7 @@ spec:
           - name: DD_CLUSTER_AGENT_AUTH_TOKEN
             valueFrom:
               secretKeyRef:
-                name: {{ template "datadog.fullname" . }}-cluster-agent
+                name: {{ if not .Values.clusterAgent.tokenExistingSecret }}{{ template "datadog.fullname" . }}-cluster-agent{{ else }}"{{ .Values.clusterAgent.tokenExistingSecret }}"{{ end }}
```

Comment:
since this line used in 3 different places, maybe it is worth to create a helm `template`.

Reply:
Thanks @clamoriniere - I realized that though I defined the `clusterAgent.tokenSecretName`, I did not make use of it. I have moved the repeated functionality to `clusterAgent.tokenSecretName` template.